cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/BrokerFailureDetector.java

```diff
@@ -223,31 +236,27 @@ private void reportBrokerFailures() {
   }
 
   /**
-   * The zookeeper listener for failure detection.
+   * The zookeeper handler for failure detection.
    */
-  private class BrokerFailureListener implements IZkChildListener {
+  private final class BrokerFailureHandler implements ZNodeChildChangeHandler {
 
-    @Override
-    public void handleChildChange(String parentPath, List<String> currentChildren) {
-      // Ensure that broker failures are not reported if there are no updates in already known failed brokers.
-      // Anomaly Detector guarantees that a broker failure detection will not be lost. Skipping reporting if not updated
-      // ensures that the broker failure detector will not report superfluous broker failures due to flaky zNode.
-      detectBrokerFailures(currentChildren.stream().map(Integer::parseInt).collect(toSet()), true);
+    private String _path;
```

Comment:
Nit: Can we make this `final`? Alternatively, maybe we can drop this variable and make `path()` always return `BrokerIdsZNode.path()`.